### PR TITLE
feat: detect armv7l-based mobile devices

### PR DESCRIPTION
**Pull request by antongolub**
@juliangruber,

Let's try again. This PR brings:

* armv7l-based devices detection
* 900 unit tests
* deps update / vuls fixes
* tablet regex optimization

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -45,3 +47,29 @@ test('is mobile', function (t) {
...
+  const testCases = Object.keys(checks).reduce((cases, deviceCategory) => {
+    cases.push(
+      ...new Array(limit).fill().map(() => ({
+        ua: new UserAgent({ deviceCategory }).toString(),
+        ...checks[deviceCategory]
+      }))
+    )
+
+    return cases
+  }, [])
```
Comment (juliangruber) [inline test.js:67]:
```suggestion
  const testCases = Object
    .entries(checks)
    .reduce((cases, [deviceCategory, { result, tablet }]) => ([
      ...cases,
      ...new Array(limit).fill().map(() => ({
        ua: new UserAgent({ deviceCategory }).toString(),
        result,
        tablet
      }))
    ]), [])
```

- Cleaner lookup using `Object.entries`
- use `...` to simplify the logic
- make explicit which properties get added

File: test.js [hunk 2]
```diff
@@ -45,3 +47,29 @@ test('is mobile', function (t) {
 
   t.end()
 })
+
+describe('ua-bruteforce', function () {
```
Comment (juliangruber) [inline test.js:51]: awesome idea to test this! Now I have confidence in your new regexp :)

File: test.js [hunk 3]
```diff
@@ -45,3 +47,29 @@ test('is mobile', function (t) {
...
+        ...checks[deviceCategory]
+      }))
+    )
+
+    return cases
+  }, [])
+
+  testCases.forEach(({ ua, result, tablet }) => {
+    test(ua, t => {
+      t.equal(isMobile({ ua, tablet }), result)
```
Comment (juliangruber) [inline test.js:71]: What about testing for false positives?

File: index.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
+/* eslint-disable no-var */
+
```
Comment (juliangruber) [inline index.js:2]: Since we're changing this already, let's switch to `const` :)

File: index.js [hunk 5]
```diff
@@ -17,7 +19,7 @@ function isMobile (opts) {
   }
   if (typeof ua !== 'string') return false
 
-  var result = opts.tablet ? tabletRE.test(ua) : mobileRE.test(ua)
+  var result = mobileRE.test(ua) || (!!opts.tablet && tabletRE.test(ua))
```
Comment (juliangruber) [inline index.js:22]:
```suggestion
  var result = mobileRE.test(ua) || (opts.tablet && tabletRE.test(ua))
```

This boolean cast doesn't change the results

**Review by juliangruber — CHANGES_REQUESTED** [review]
Awesome, this is progressing in a great way!

File: index.js (hunk 4 above)
Comment (antongolub) [reply index.js:2]: Ok. I just thought es5 support is still needed.

File: test.js (hunk 3 above)
Comment (antongolub) [reply test.js:71]: Hmm ... Doesn't the absence of errors in each device category guarantee the correctness of the impl? No test detects that `desktop` is identified as `mobile` device, or vice versa.

File: test.js (hunk 1 above)
Comment (antongolub) [reply test.js:67]: Agree. This is really much better.

File: index.js (hunk 4 above)
Comment (juliangruber) [reply index.js]: This will be a breaking change so we can just stop supporting it

File: index.js (hunk 5 above)
Comment (juliangruber) [reply index.js]: Right!

File: test.js (hunk 3 above)
Comment (juliangruber) [reply test.js:72]: You're right, the `desktop` category takes care of this

**Comment by juliangruber** [comment]
https://github.com/juliangruber/is-mobile/releases/tag/v3.0.0

🥳

**Comment by juliangruber** [comment]
https://github.com/juliangruber/is-mobile/commit/1fa47ecba908551b9f606ff1a813e616a4aea6e5

**Pull request merged**